libs/SmartSync/SmartSync/Classes/Util/SFSoqlSyncDownTarget.m

```diff
@@ -147,27 +150,27 @@ - (void) getListOfRemoteIds:(SFSmartSyncSyncManager*)syncManager
     NSString* fromClause = [self.query substringFromIndex:rangeFirst.location];
     [soql appendString:fromClause];
     __block NSUInteger countFetched = 0;
-    __block NSUInteger totalSize = 0;
     __block NSMutableArray* allRecords = [[NSMutableArray alloc] init];
-    __block SFSyncDownTargetFetchCompleteBlock completionBlockRecurse = ^(NSArray *records) {};
-    __weak SFSoqlSyncDownTarget* weakSelf = self;
-    SFSyncDownTargetFetchCompleteBlock completionBlock = ^(NSArray* records) {
-        totalSize = self.totalSize;
+    __block SFSyncDownTargetFetchCompleteBlock fetchBlockRecurse = ^(NSArray *records) {};
```

Comment:
In addition to my previous comments, it wouldn't be bad for this to be a `__weak` reference as well, as it *is* possible to capture the block in a retain cycle with this pattern, otherwise.